pandas/core/arrays/masked.py

```diff
@@ -1093,7 +1093,13 @@ def _mode(self, dropna: bool = True) -> Self:
             result = mode(self._data, dropna=dropna, mask=self._mask)
             res_mask = np.zeros(result.shape, dtype=np.bool_)
         else:
-            result, res_mask = mode(self._data, dropna=dropna, mask=self._mask)
+            res_tuple = mode(self._data, dropna=dropna, mask=self._mask)
```

Comment:
Is it possible to make the return type of `mode` consistent throughout the code base so we don't have a check like this?

Reply:
I did a little investigation earlier and `res_mask` seems to be incorrect here if the series has `pd.NA` in it.

Reply:
With the `mode` return type consistency, it's called in a bunch of other places using the return with no mask, so not sure if it's worth changing it to then have to uglify other usages to `result, _ = mode(...)`, `return mode(...)[0]`

Reply:
I agree it's good to have a consistent return type. `result, _ = mode(...)` is not ugly to me.